x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -623,7 +637,9 @@ export function App({
         )}
       </div>
       {lastKnownDoc && state.isSaveModalVisible && (
-        <SavedObjectSaveModalOrigin
+        <TagEnhancedSavedObjectSaveModalOrigin
+          savedObjectsTagging={savedObjectsTagging}
+          initialTags={tagsIds}
```

Comment:
I started by trying to inline the enhancement in the app, but that was a pain to code and to read, so I used a HOC around `SavedObjectSaveModalOrigin` instead. the HOC currently lives in the `lens` app, but I could easily see it move to the so tagging plugin once we have more consumers for it.